api/diagnostics.go

```diff
@@ -219,17 +219,21 @@ func (j *DiagnosticsJob) runBackgroundJob(nodes []Node, cfg *config.Config, DCOS
 	// place a summaryErrorsReport.txt in a zip archive which should provide info what failed during the logs collection.
 	summaryErrorsReport := new(bytes.Buffer)
 	defer func() {
-		zipFile, err := zipWriter.Create("summaryErrorsReport.txt")
-		if err != nil {
-			j.Status = "Could not append a summaryErrorsReport.txt to a zip file"
-			logrus.Errorf("%s: %s", j.Status, err)
-			j.Errors = append(j.Errors, err.Error())
-			return
+		// add a summaryErrorsReport.txt file to a diagnostics bundle, if it's not empty
+		if summaryErrorsReport.Len() > 0 {
+			zipFile, err := zipWriter.Create("summaryErrorsReport.txt")
+			if err != nil {
+				j.Status = "Could not append a summaryErrorsReport.txt to a zip file"
+				logrus.Errorf("%s: %s", j.Status, err)
+				j.Errors = append(j.Errors, err.Error())
+				return
+			}
+
+			io.Copy(zipFile, summaryErrorsReport)
```

Comment:
we should probably not ignore the error from `io.Copy` either.

Reply:
good catch, fixed